### DISCUSSION TITLE
configuring Ray for use with Laradock

### DISCUSSION THREAD

**Discussion by jayenne** [spatie/ray#120] [Q&A]
Hey Spatie, Hay all o/

it may be of use to some to include a secontion the docks for seting up Ray with Laradock as it's really simple..

* Follow the instructions for setting Ray up with docker [here](https://spatie.be/docs/ray/v1/configuration/docker) (saving the ray.php in your project's root folder).
* open your `laradock/docker-compose.yml` file, search for `extra_hosts:` thne add another line with the new `- "host.docker.internal:host-gateway"` entry under the existing `- "dockerhost:${DOCKER_HOST_IP}"` 

FYI: your docker-compose.yml'z extra_hosts should now look like:
```
extra_hosts:
        - "dockerhost:${DOCKER_HOST_IP}"
        - "host.docker.internal:host-gateway"
```

* Restart docker: `docker-compose restart`

Try, Buy, Fly!

**Comment by freekmurze** [comment]
> Try, Buy, Fly!

Lol!

Feel free to PR any extra docs to this page (there's an edit button at the top)
https://spatie.be/docs/ray/v1/configuration/docker

**Comment by victoralbino** [comment]
> Ei Spatie, Hay all o /
> 
> pode ser útil para alguns incluir uma seção das docas para configurar Ray com Laradock, pois é muito simples.
> 
> * Siga as instruções para configurar o Ray com docker [aqui](https://spatie.be/docs/ray/v1/configuration/docker) (salvando o ray.php na pasta raiz do seu projeto).
> * abra seu `laradock/docker-compose.yml`arquivo, pesquise `extra_hosts:`e adicione outra linha com a nova `- "host.docker.internal:host-gateway"`entrada sob o existente`- "dockerhost:${DOCKER_HOST_IP}"`
> 
> Para sua informação: seu docker-compose.yml'z extra_hosts agora deve ser semelhante a:
> 
> ```
> extra_hosts:
>         - "dockerhost:${DOCKER_HOST_IP}"
>         - "host.docker.internal:host-gateway"
> ```
> 
> * Reinicie o docker: `docker-compose restart`
> 
> Experimente, compre, voe!

Using your solution I got the following error:

```
ERROR: for laradock_workspace_1  Cannot create container for service workspace: invalid IP address in add-host: "host-gateway"

ERROR: for workspace  Cannot create container for service workspace: invalid IP address in add-host: "host-gateway"

```

**Comment by jayenne** [comment]
Did you follow the docker install instructions? Which OS are you using?
Please let us know the developoment envoronment and the steps required to reproduce your error

**Reply by victoralbino** [reply]
I am using the default laradocker configuration for multiple applications. https://laradock.io/getting-started/#B
Operating System: Ubuntu 20.04.1 LTS.

I even reinstalled the laradocker, but the same error

**Comment by victoralbino** [comment]
I discovered my mistake, it was to include it inside the docker workspace, but it must be included within php-fpm

**Reply by jayenne** [reply]
I'm happy you have it working. I don't understand what you've put in php-fpm? what is that?
*obviously knowing this may be very helpful to know you others exeriencing difficulties.
